compiler/noirc_frontend/src/elaborator/types.rs

```diff
@@ -1808,6 +1819,16 @@ impl<'context> Elaborator<'context> {
     }
 }
 
+fn lookup_method_in_trait(
+    the_trait: &Trait,
+    method_name: &str,
+    trait_bound: &ResolvedTraitBound,
+) -> Option<HirMethodReference> {
+    let trait_method = the_trait.find_method(method_name)?;
+    let generics = trait_bound.trait_generics.clone();
```

Comment:
The generics here could be wrong in the case of an inherited trait bound with generics. E.g. for the trait

```noir
trait Foo<A>: Bar<A> { ... }

trait Bar<T> { fn bar(self); }
```
When looking up `bar` (in `lookup_method_in_trait_constraints`) from a trait bound of `Foo<B>` we'd use the trait definition `Bar<T>` and the generics list would be `T` instead of the desired `B`

Reply:
I see... yes, to be honest, I get a bit lost with generics here. But I'll try to figure it out. I guess it'll get trickier if a trait has a parent that also has a parent, etc.

Reply:
I think I figured it out :-)